### PR TITLE
Add Redox OS CI

### PR DESCRIPTION
**Pull request by jackpot51**
This adds Redox OS builds to CI. Tests are skipped at the moment.

Part of https://github.com/uutils/coreutils/issues/5210

### CODE REVIEW

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -525,7 +526,7 @@ jobs:
           i686-*) TARGET_ARCH=i686 ;;
           x86_64-*) TARGET_ARCH=x86_64 ;;
         esac;
-        unset TARGET_OS ; case '${{ matrix.job.target }}' in *-linux-*) TARGET_OS=linux ;; *-apple-*) TARGET_OS=macos ;; *-windows-*) TARGET_OS=windows ;; esac;
+        unset TARGET_OS ; case '${{ matrix.job.target }}' in *-linux-*) TARGET_OS=linux ;; *-apple-*) TARGET_OS=macos ;; *-windows-*) TARGET_OS=windows ;; *-redox*) TARGET_OS=redox ;; esac;
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:529]: maybe update this block to be on multiple lines now that we have a few ?

File: .github/workflows/CICD.yml [hunk 2]
```diff
@@ -560,8 +561,11 @@ jobs:
         if [ -n "${{ matrix.job.features }}" ]; then CARGO_FEATURES_OPTION='--features=${{ matrix.job.features }}' ; fi
         outputs CARGO_FEATURES_OPTION
         # * CARGO_CMD
-        CARGO_CMD='cross' ; case '${{ matrix.job.use-cross }}' in ''|0|f|false|n|no) CARGO_CMD='cargo' ;; esac;
+        CARGO_CMD='cross' ; case '${{ matrix.job.use-cross }}' in ''|0|f|false|n|no) CARGO_CMD='cargo' ;; redoxer) CARGO_CMD='redoxer' ;; esac;
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:564]: same

Comment (jackpot51) [reply .github/workflows/CICD.yml]: Formatted in latest commit

File: .github/workflows/CICD.yml (hunk 1 above)
Comment (jackpot51) [reply .github/workflows/CICD.yml]: Formatted in latest commit

File: .github/workflows/CICD.yml [hunk 3]
```diff
@@ -525,7 +526,13 @@ jobs:
           i686-*) TARGET_ARCH=i686 ;;
           x86_64-*) TARGET_ARCH=x86_64 ;;
         esac;
-        unset TARGET_OS ; case '${{ matrix.job.target }}' in *-linux-*) TARGET_OS=linux ;; *-apple-*) TARGET_OS=macos ;; *-windows-*) TARGET_OS=windows ;; esac;
+        unset TARGET_OS
+        case '${{ matrix.job.target }}' in
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:530]: much better, thanks

**Comment by sylvestre** [comment]
@jackpot51 we are going to apply to the gsoc
https://uutils.github.io/gsoc.html
we had:
```
Official Redox support

We want to support the Redox operating system, but are not actively testing against it. Since the last round of fixes in [#2550](https://github.com/uutils/coreutils/pull/2550), many changes have probably been introduced that break Redox support. This project would involve setting up Redox in the CI and fixing any issues that arise and porting features over.

    Difficulty: Medium
    Size: 175 hours
    Mentors: TBD
    Required skills:
        Rust
```

is there enough work for a gsoc ?  thanks

**Comment by jackpot51** [comment]
@sylvestre I think there would be, I'd recommend to joim our matrix to discuss: https://redox-os.org/community/

**Pull request merged**